### PR TITLE
chore: migrate http server to ts

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: utils/testserver/index.ts [hunk 1]
```diff
@@ -14,65 +15,67 @@
...
-   * @param {!Object=} sslOptions
-   */
-  constructor(dirPath, port, loopback, sslOptions) {
+  constructor(dirPath: string, port: number, loopback?: string, sslOptions?: object) {
     if (sslOptions)
-      this._server = https.createServer(sslOptions, this._onRequest.bind(this));
+      this._server = _createServer(sslOptions, this._onRequest.bind(this));
     else
-      this._server = http.createServer(this._onRequest.bind(this));
+      this._server = createServer(this._onRequest.bind(this));
```
Comment (dgozman) [inline utils/testserver/index.ts:74]: `http.createServer` would be much more readable :)

File: utils/testserver/index.ts [hunk 2]
```diff
@@ -14,65 +15,67 @@
...
-   * @param {string} dirPath
-   * @param {number} port
-   * @param {string=} loopback
-   * @param {!Object=} sslOptions
-   */
-  constructor(dirPath, port, loopback, sslOptions) {
+  constructor(dirPath: string, port: number, loopback?: string, sslOptions?: object) {
     if (sslOptions)
-      this._server = https.createServer(sslOptions, this._onRequest.bind(this));
+      this._server = _createServer(sslOptions, this._onRequest.bind(this));
```
Comment (dgozman) [inline utils/testserver/index.ts:72]: `https.createServer`?

File: utils/testserver/index.ts [hunk 3]
```diff
@@ -14,65 +15,67 @@
...
-const mime = require('mime');
-const WebSocketServer = require('ws').Server;
+import fs from 'fs';
+import url from 'url';
+import type http from 'http';
+import { createServer } from 'http';
+import { createServer as _createServer } from 'https';
+import { getType } from 'mime';
+import type { Socket } from 'net';
+import { join } from 'path';
```
Comment (dgozman) [inline utils/testserver/index.ts:25]:
```suggestion
import path from 'path';
```

File: utils/testserver/index.ts [hunk 4]
```diff
@@ -312,7 +262,7 @@ class TestServer {
       return;
     }
     const extension = filePath.substring(filePath.lastIndexOf('.') + 1);
-    const mimeType = mime.getType(extension) || 'application/octet-stream';
+    const mimeType = getType(extension) || 'application/octet-stream';
```
Comment (dgozman) [inline utils/testserver/index.ts:265]: `mime.getType()`?

**Review by dgozman — APPROVED** [review]

File: utils/testserver/index.ts (hunk 1 above)
Comment (pavelfeldman) [reply utils/testserver/index.ts:74]: You dont say!

File: utils/testserver/index.ts (hunk 2 above)
Comment (pavelfeldman) [reply utils/testserver/index.ts:72]: You dont say!

File: utils/testserver/index.ts (hunk 3 above)
Comment (pavelfeldman) [reply utils/testserver/index.ts:25]: You dont say!

File: utils/testserver/index.ts (hunk 4 above)
Comment (pavelfeldman) [reply utils/testserver/index.ts:265]: You dont say!

**Pull request merged**